pg_wait_sampling.c

```diff
@@ -501,7 +515,7 @@ pg_wait_sampling_get_current(PG_FUNCTION_ARGS)
 			{
 				PGPROC *proc = &ProcGlobal->allProcs[i];
 
-				if (proc != NULL && proc->pid != 0 && proc->wait_event_info)
+				if (proc != NULL && proc->pid != 0 && (proc->wait_event_info || pgws_collector_hdr->sampleCpu))
```

Comment:
I think we need the latch owner check here too. Maybe it's better to replace a single `if` with a couple of `if-continue` statements like in the collector, for symmetry.

Reply:
Because it was basically the same as probe_waits() I added a commit to factor it into a separate function.